compiler/rustc_resolve/src/late.rs

```diff
@@ -1841,8 +1841,9 @@ impl<'a: 'ast, 'b, 'ast> LateResolutionVisitor<'a, 'b, 'ast> {
 
             // Before we start looking for candidates, we have to get our hands
             // on the type user is trying to perform invocation on; basically:
-            // we're transforming `HashMap::new` into just `HashMap`
-            let path = if let Some((_, path)) = path.split_last() {
+            // we're transforming `HashMap::new` into just `HashMap`.
+            let path = if path.len() >= 2 {
+                let (_, path) = path.split_last().unwrap();
```

Comment:
How about `path.split_last().unwrap().1`?

Reply:
I'd prefer to avoid an unwrap and keep the match, e.g.
```rust
match path.split_last() {
    ((_, path)) if !path.is_empty() => path,
    _ => return ...
}
```

Reply:
Avoiding unwrap makes sense. Will update.